hardware/src/main/java/at/amartinz/hardware/utils/HwIoUtils.java

```diff
@@ -213,6 +213,40 @@ public static boolean fileExists(final String[] files) {
         return ((content != null) ? content.trim() : null);
     }
 
+	/**
+     * Read one line from a file with root credentials.
+     * Use blocking reading.
+     * @param Path of the file to read
+     * @return Line read.
+     */
+    @WorkerThread @Nullable public static String readOneLineRoot(final String path) {
+        final String content = readFileInternalRoot(path, true);
+        return ((content != null) ? content.trim() : null);
+    }
+
+	/**
+	 * Read file with root permissions.
+     * @param path Path of the file to read
+     * @param oneLine read only one line if true
+     * @return content read
+     */
+    @WorkerThread
+    @Nullable
+    public static String readFileInternalRoot(@Nullable final String path, boolean oneLine) {
+        if (TextUtils.isEmpty(path) || !RootCheck.isRooted()) {
+            return null;
+        }
+        String output = null;
+        if(oneLine)
+            output = RootShell.fireAndBlockStringNewline(
+                    new Command(String.format("head -n 1 %s", path)));
```

Comment:
I'm not familiar with android roms and busybox yet, but it seems to be a busybox and toybox command :
https://busybox.net/downloads/BusyBox.html
https://www.landley.net/toybox/help.html
However all busybox commands are not in Android. Wikipedia says Toybox is included since android 6

I download a stock 4.4 android stock rom for x86 (from https://www.fosshub.com/Android-x86.html) and it has head. I looked on a 2.3.2 phone and it doesn't. 

I don't know much how to check further. I think it is shipped with at least all 4.4 distros.
Or I can use cat and extract the first line, but that may not be the cleaner option.